nativeunwind/elfunwindinfo/elfehframe_x86.go

```diff
@@ -174,27 +174,73 @@ func detectEntryX86(code []byte) int {
 	// On musl, the entry has no FDE, or possibly has an FDE covering part of it.
 	// Detect the musl case and return entry.
 
-	// Match the assembly exactly except the LEA call offset
-	if len(code) < 32 ||
-		!bytes.Equal(code[:9], []byte{0x48, 0x31, 0xed, 0x48, 0x89, 0xe7, 0x48, 0x8d, 0x35}) ||
-		!bytes.Equal(code[13:22], []byte{0x48, 0x83, 0xe4, 0xf0, 0xe8, 0x00, 0x00, 0x00, 0x00}) {
+	// Expected pattern:
+	//   xor    %rbp,%rbp
+	//   mov    %rsp,%rdi
+	//   lea    <offset>(%rip),%rsi
+	//   and    $0xfffffffffffffff0,%rsp
+	//   call   <offset>
+	//   ... (more instructions)
+	//   jmp    <offset>
+
+	if len(code) < 32 {
+		return 0
+	}
+
+	interp := amd.NewInterpreterWithCode(code)
+	pos := 0
+
+	// First instruction: xor %rbp,%rbp
+	inst, err := interp.Step()
+	if err != nil || inst.Op != x86asm.XOR {
+		return 0
+	}
+	pos += inst.Len
+
+	// Second instruction: mov %rsp,%rdi
+	inst, err = interp.Step()
+	if err != nil || inst.Op != x86asm.MOV {
```

Comment:
This is checking only opcode, not operands. Same applies for rest of the checks.